src/rdata/naptr.rs

```diff
@@ -0,0 +1,545 @@
+        self.order.hash(state);
+        self.preference.hash(state);
+        self.flags.hash(state);
+        self.services.hash(state);
+        self.regexp.hash(state);
+        self.replacement.hash(state);
+    }
+}
+
+//--- RecordData, ParseRecordData, ComposeRecordData
+
+impl<Octs, Name> RecordData for Naptr<Octs, Name> {
+    fn rtype(&self) -> Rtype {
+        Rtype::NAPTR
+    }
+}
+
+impl<'a, Octs: Octets + ?Sized> ParseRecordData<'a, Octs>
+    for Naptr<Octs::Range<'a>, ParsedName<Octs::Range<'a>>>
+{
+    fn parse_rdata(
+        rtype: Rtype,
+        parser: &mut Parser<'a, Octs>,
+    ) -> Result<Option<Self>, ParseError> {
+        if rtype == Naptr::RTYPE {
+            Self::parse(parser).map(Some)
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+impl<Octs, Name> ComposeRecordData for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    Name: ToName,
+{
+    fn rdlen(&self, _compress: bool) -> Option<u16> {
+        Some(
+            2 + 2
+                + self.flags.compose_len()
+                + self.services.compose_len()
+                + self.regexp.compose_len()
+                + self.replacement.compose_len(),
+        )
+    }
+
+    fn compose_rdata<Target: crate::base::wire::Composer + ?Sized>(
+        &self,
+        target: &mut Target,
+    ) -> Result<(), Target::AppendError> {
+        self.order.compose(target)?;
+        self.preference.compose(target)?;
+        self.flags.compose(target)?;
+        self.services.compose(target)?;
+        self.regexp.compose(target)?;
+        self.replacement.compose_canonical(target)
+    }
+
+    fn compose_canonical_rdata<
+        Target: crate::base::wire::Composer + ?Sized,
+    >(
+        &self,
+        target: &mut Target,
+    ) -> Result<(), Target::AppendError> {
+        self.compose_rdata(target)
+    }
+}
+
+//--- Display
+
+impl<Octs, Name> core::fmt::Display for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    Name: fmt::Display,
+{
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        write!(
+            f,
+            "{} {} {} {} {} {}.",
+            self.order,
+            self.preference,
+            self.flags.display_quoted(),
+            self.services.display_quoted(),
+            self.regexp.display_quoted(),
+            self.replacement
+        )
+    }
+}
+
+//--- Debug
+
+impl<Octs, Name> core::fmt::Debug for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    Name: fmt::Debug,
+{
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        f.debug_struct("Naptr")
+            .field("order", &self.order)
+            .field("preference", &self.preference)
+            .field("flags", &self.flags)
+            .field("services", &self.services)
+            .field("regexp", &self.regexp)
+            .field("replacement", &self.replacement)
+            .finish()
+    }
+}
+
+//--- ZonefileFmt
+
+impl<Octs, Name> ZonefileFmt for Naptr<Octs, Name>
```

Comment:
@tertsdiepraam: Can you review this fn please?

Reply:
Looks good!